projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -781,15 +781,22 @@ function wpcom_site_has_global_styles_in_personal_plan( $blog_id = 0 ) {
 		return false;
 	}
 
-	if ( ! function_exists( '\ExPlat\assign_given_user' ) ) {
+	if ( ! function_exists( '\ExPlat\assign_given_user' ) || ! function_exists( '\WPCOM\Experiments\Internal\get_cached_experiment_by_name' ) ) {
+		return false;
+	}
+
+	// If the experiment is not running and the request isn't proxied, we can return early.
+	$exp     = \WPCOM\Experiments\Internal\get_cached_experiment_by_name( 'calypso_plans_global_styles_personal_20240127' );
```

Comment:
Can we shortcircuit this function call behind that proxied variable? Seems that the cache is filesystem-based, so it's not that lightweight.

Reply:
We need this applied to all users not just proxied users. If the experiment isn't running we do not want to assign a group.

But I'm thinking that I'll just change the approach and change the cache key. It's going to be simpler but we need to merge once the experiment starts.